src/cryptenroll/cryptenroll-tpm2.c

```diff
@@ -208,25 +209,52 @@ int enroll_tpm2(struct crypt_device *cd,
                         return log_debug_errno(r, "Failed to read TPM PCR signature: %m");
         }
 
+        bool any_pcr_value_specified = tpm2_pcr_values_has_any_values(hash_pcr_values, n_hash_pcr_values);
+
         _cleanup_(tpm2_pcrlock_policy_done) Tpm2PCRLockPolicy pcrlock_policy = {};
         if (pcrlock_path) {
                 r = tpm2_pcrlock_policy_load(pcrlock_path, &pcrlock_policy);
                 if (r < 0)
                         return r;
 
+                any_pcr_value_specified = true;
                 flags |= TPM2_FLAGS_USE_PCRLOCK;
         }
 
         _cleanup_(tpm2_context_unrefp) Tpm2Context *tpm2_context = NULL;
-        r = tpm2_context_new(device, &tpm2_context);
-        if (r < 0)
-                return log_error_errno(r, "Failed to create TPM2 context: %m");
+        TPM2B_PUBLIC device_key_public = {};
+        if (device_key) {
+                _cleanup_free_ char *device_key_buffer = NULL;
+                size_t device_key_buffer_size;
+                r = read_full_file(device_key, &device_key_buffer, &device_key_buffer_size);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read device key from file: %m");
+
+                TSS2_RC rc;
+                size_t offset = 0;
+                rc = sym_Tss2_MU_TPM2B_PUBLIC_Unmarshal(
+                                (uint8_t*) device_key_buffer,
+                                device_key_buffer_size,
+                                &offset,
+                                &device_key_public);
+                if (rc != TSS2_RC_SUCCESS)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "Could not unmarshal public key from file.");
 
```

Comment:
btw, i think here we might want to check that offset macthes device_key_buffer_size to disallow trailing garbage

Reply:
ack, added